### ISSUE TITLE
Confirmation modal of bulk-select/submit before final submit to backend

### ISSUE THREAD

**Issue by yuanzhou** [x-atlas-consortia/data-ingest-board#184]
This is an enhancement to improve the user experience following the initial implementation of https://github.com/x-atlas-consortia/data-ingest-board/issues/173. 

<img width="982" alt="Screenshot 2024-04-05 at 1 56 15 PM" src="https://github.com/x-atlas-consortia/data-ingest-board/assets/195873/2f75adbd-d6f2-4fb7-a65f-f4bdb8761acb">

Upon clicking the "Submit For Processing" button, it'll be very helpful if we display a confirmation modal/table with all the selected records (`hubmap_id`, `status`, and `group_name` should suffice). Then the user clicks a "Submit" button to confirm or "Cancel" button to go back to edit the list if anything turns out to be selected by mistake or some records might be missed. 

Because after selecting across different pages or filtering, it can be hard to keep tracking what's in the list to submit. This final step will help the users to confirm and possibly minimize operator errors.

**Comment by libpitt** [comment]
I find some inconsistent and confusing UI behaviors between select/deselect/filter updates. Attached two videos.

From @yuanzhou 

- In the 1st video, I selected two QA records then filtered by "Incomplete" status, the previously selected 2 records still there by then. But when I selected the additional 3 Incomplete records by clicking the checkbox in header, the 2 previous selected records disappeared.
- In the 2nd video, I selected 3 Incomplete records first, then filtered by "New". Upon selecting one single record row rather than bulk select checkbox in header, all the 3 previously selected records disappeared too.

**I had thought all the selected records would be appended to the bulk-select list and stay there regardless of filtering and additional selections, and we only reset the bulk list after I click the "Submit For Processing" button.**

In addition, I also think it'll be very helpful if we display a confirmation modal/table with all the selected records (hubmap_id, status, and group should suffice). Then the user clicks a "Submit" button to confirm or "Cancel" button to go back to edit the list if anything turns out to be selected by mistake or some records might be missed. Because after selecting across different pages or filtering, it's hard to keep tracking what's in the list to submit.
